packages/cliqz-home-react/src/SpeedDialsRow.js

```diff
@@ -0,0 +1,160 @@
+import React, { Component } from 'react';
+import SpeedDial from './SpeedDial';
+import AddSpeedDial from './AddSpeedDial';
+import cliqz from './Cliqz';
+
+class SpeedDialsRow extends Component {
+  constructor(props) {
+    super(props);
+
+    // this.closeUndo = this.closeUndo.bind(this);
+    this.undoRemoval = this.undoRemoval.bind(this);
+    this.resetAll = this.resetAll.bind(this);
+  }
+
+  componentWillMount() {
+    this.state = {
+      isCustom: this.props.type === 'custom',
+      showAddButton: () => {
+        return (this.state.isCustom && this.state.displayAddBtn());
+      },
+      displayAddBtn: () => {
+        return this.props.dials.length < 5;
+      },
+      removedItems: []
+    }
+  }
+
+  removeSpeedDial(dial, index) {
+    this.props.removeSpeedDial(dial, index);
+    dial.removedAt = index;
+    this.state.removedItems.push(dial);
```

Comment:
not sure if we should change a state like this. this is probably more idiomatic:
```
this.setState({
  removedItems: [
    ...this.state.removedItems,
    dial,
  ],
});